docs/reference/updating-mapper.md

```diff
@@ -0,0 +1,179 @@
+# Updating the doc mapping of an index
+
+Quickwit allows updating the mapping it uses, to add more fields to an existing index, or change how they are indexed. In doing so,
+it does not reindex existing data, still lets you search through older documents where possible.
+
+## Indexing
+
+When you update a doc mapping for an index, Quickwit will restart indexing pipelines to take the changes into account. As both this operation and the document ingestion are asynchronous, you can't assume documents sent immediately after the update will necessarily use the new mapping nor that documents sent immediately before the update won't be indexed using the new doc mapping.
+
+## Querying
+
+When receiving a query, Quickwit always validate it against the most recent mapping.
+If a query was valid under a previous mapping but is not compatible with the newer mapping, that query will be rejected.
+For instance if a field which was indexed no longer is, any query that uses it will become invalid.
+On the other hand, if a query was not valid for a previous doc mapping, but is valid under the new doc mapping, Quickwit will process the query.
+When querying newer splits, it will behave normally, when querying older splits, it will try ot execute the query as correctly as possible.
+If you find a situation where older splits causes a valid request to return an error, please open a bug report.
+See example 1 and 2 below for clarification.
+
+Change in tokenizer affect only newer splits, older splits keep using the tokenizers they were created with.
+
+Document retrieved are mapped from Quickwit internal format to JSON based on the latest doc mapping. This means if fields are deleted,
+they will stop appearing (see also Reversibility below) unless mapper mode is Dynamic. If the type of some field changed, it will be converted on a best effort basis:
```

Comment:
```suggestion
they will stop appearing (see also Reversibility below) unless mapper mode is Dynamic. If the type of some field changed, it will be converted on a best-effort basis:
```